server/lib/budget.js

```diff
@@ -221,37 +248,267 @@ export async function getTotalAmountPaidExpenses(collective, { startDate, endDat
+      transactionType: CREDIT,
+    });
+
+    result = results[collective.id];
+  }
+
+  return { contributions: result.count, contributors: result.countDistinctFromCollective };
+}
+
+export async function countTransactionsAndDistinctFromCollectives(
+  ids,
+  {
+    startDate = null,
+    endDate = null,
+    excludeRefunds = true,
+    hostCollectiveId = null,
+    excludeInternals = false,
+    includeChildren = false,
+    transactionType = null,
+    kind = null,
+  } = {},
+) {
+  const include = [];
+
+  const where = {};
+
+  if (ids) {
+    if (includeChildren) {
+      include.push({
+        model: models.Collective,
+        as: 'collective',
+        where: {
+          [Op.or]: {
+            id: ids,
+            ParentCollectiveId: ids,
+          },
+        },
+        attributes: [],
+      });
+      // TODO: exclude internal transactions to avoid double counting
+      // include.push({
+      //   model: models.Collective,
+      //   as: 'fromCollective',
+      //   // where: {
+      //   //   [Op.not]: {
+      //   //     [Op.and]: {
+      //   //     id: ids,
+      //   //     ParentCollectiveId: ids,
+      //   //     },
+      //   //   },
+      //   // },
+      //   attributes: [],
+      // });
+    } else {
+      where.CollectiveId = ids;
+    }
+  }
+  if (transactionType) {
+    where.type = transactionType;
+  }
+
+  if (startDate) {
+    where.createdAt = where.createdAt || {};
+    where.createdAt[Op.gte] = startDate;
+  }
+  if (endDate) {
+    where.createdAt = where.createdAt || {};
+    where.createdAt[Op.lt] = endDate;
+  }
+  if (excludeRefunds) {
+    // Exclude refunded transactions
+    where.RefundTransactionId = { [Op.is]: null };
+    // Also exclude anything with isRefund=true (PAYMENT_PROCESSOR_COVER doesn't have RefundTransactionId set)
+    where.isRefund = { [Op.not]: true };
+  }
+
+  if (hostCollectiveId) {
+    // Only transactions that are marked under a Fiscal Host
+    where.HostCollectiveId = hostCollectiveId;
+  }
+  if (excludeInternals) {
+    // Exclude internal transactions (we can tag some Transactions like "Switching Host" as internal)
+    where.isInternal = { [Op.not]: true };
+  }
+  if (kind) {
+    where.kind = kind;
+  }
+
+  const totals = {};
+
+  // Initialize totals
+  if (ids) {
+    for (const CollectiveId of ids) {
+      totals[CollectiveId] = { CollectiveId, count: 0, countDistinctFromCollective: 0 };
+    }
+  }
+
+  const CollectiveId = includeChildren
+    ? [
+        sequelize.fn('COALESCE', sequelize.col('collective.ParentCollectiveId'), sequelize.col('collective.id')),
+        'CollectiveId',
+      ]
+    : sequelize.col('CollectiveId');
+
+  const attributes = [
+    CollectiveId,
+    [sequelize.fn('COUNT', sequelize.col('Transaction.id')), 'count'],
+    [
+      sequelize.fn('COUNT', sequelize.fn('DISTINCT', sequelize.col('Transaction.FromCollectiveId'))),
+      'countDistinctFromCollective',
+    ],
```

Comment:
In case we're doing NET_CREDIT in the future, each host fee will count as a transaction and open collective would be a contributor. But that's not the case in this version.